pallets/rewards/src/lib.rs

```diff
@@ -230,11 +235,22 @@ pub mod pallet {
 			Groups::<T, I>::try_mutate(group_id, |group| {
 				let reward_to_mint = T::RewardMechanism::reward_group(group, reward)?;
 
-				T::Currency::mint_into(
-					T::RewardCurrency::get(),
-					&T::PalletId::get().into_account_truncating(),
-					reward_to_mint,
-				)?;
+				// TODO: Check with @Luis whether we want to move this into a trait
+				if let Some(pallet_id) = T::RewardSource::get() {
+					T::Currency::transfer(
+						T::RewardCurrency::get(),
+						&pallet_id.into_account_truncating(),
+						&T::PalletId::get().into_account_truncating(),
+						reward,
+						true,
+					)?;
+				} else {
+					T::Currency::mint_into(
+						T::RewardCurrency::get(),
+						&T::PalletId::get().into_account_truncating(),
+						reward,
+					)?;
+				}
```

Comment:
I like it as it is 🚀.

Because this is something independent from block rewards, maybe we could create an isolated PR with this change (with some test) and merge it before. Wdyt?